cypress/integration/ete/registration/register.spec.ts

```diff
@@ -119,182 +95,156 @@ describe('Registration flow', () => {
-          cy.visit(`/welcome/${token}`);
-          cy.contains('Create password');
-          cy.mailosaurDeleteMessage(id);
-        },
-      );
+    cy.checkForEmailAndGetDetails(
+      unregisteredEmail,
+      timeRequestWasMade,
+      /welcome\/([^"]*)/,
+    ).then(({ body, token }) => {
+      expect(body).to.have.string('Complete registration');
+      expect(body).to.have.string('returnUrl=' + decodedReturnUrl);
+      expect(body).to.have.string('ref=' + decodedRef);
+      expect(body).to.have.string('refViewId=' + refViewId);
+      cy.visit(`/welcome/${token}`);
+      cy.contains('Create password');
     });
   });
 
   it('sends user an account exists email for user with existing account with password trying to register, clicks sign in, taken to /signin', () => {
-    cy.visit('/register');
-    const timeRequestWasMade = new Date();
-
-    cy.get('input[name=email]').type(existingWithPassword.email);
-    cy.get('[data-cy="register-button"]').click();
-
-    cy.contains('Email sent');
-    cy.contains(existingWithPassword.email);
-    cy.contains('Resend email');
-    cy.contains('Change email address');
-
-    cy.mailosaurGetMessage(
-      unregisteredAccount.serverId,
-      {
-        sentTo: existingWithPassword.email,
-      },
-      {
-        receivedAfter: timeRequestWasMade,
-      },
-    ).then((email) => {
-      cy.getEmailDetails(email).then(({ links, body, id }) => {
-        expect(body).to.have.string('This account already exists');
-        expect(body).to.have.string('Sign in');
-        expect(body).to.have.string('Reset password');
-
-        expect(links.length).to.eq(3);
-
-        const signInLink = links.find((link) => link.text === 'Sign in');
-
-        expect(signInLink).not.to.be.undefined;
-        expect(signInLink?.href ?? '').to.include('/signin');
-
-        const signInUrl = new URL(signInLink?.href ?? '');
-
-        cy.visit(signInUrl.pathname);
-        cy.url().should('include', '/signin');
-
-        cy.mailosaurDeleteMessage(id);
-      });
-    });
-  });
-
-  it('sends user an account exists email for user with existing account with password trying to register, clicks reset password on email', () => {
-    cy.visit('/register');
-    const timeRequestWasMade = new Date();
-
-    cy.get('input[name=email]').type(existingWithPassword.email);
-    cy.get('[data-cy="register-button"]').click();
-
-    cy.contains('Email sent');
-    cy.contains(existingWithPassword.email);
-    cy.contains('Resend email');
-    cy.contains('Change email address');
-
-    cy.mailosaurGetMessage(
-      unregisteredAccount.serverId,
-      {
-        sentTo: existingWithPassword.email,
-      },
-      {
-        receivedAfter: timeRequestWasMade,
-      },
-    ).then((email) => {
-      cy.getEmailDetails(email, /\/reset-password\/([^"]*)/).then(
-        ({ links, body, id, token }) => {
+    cy.createTestUser({
+      password: 'test_password',
```

Comment:
Is it worth using a random strong password here which we don't know the value of?

maybe even something simple like a randomly generated uuid (`window.crypto.randomUUID()`)  value as the password for each test.

Just thinking about potential security implications of having test users floating around with weak passwords, even in a code/dev environment 🤔

Reply:
good idea, not too hard to implement and potentially good payoff if it stops a potential vuln

Reply:
added in latest commit